### PR TITLE
ci: Fix Postgres and MySQL tests (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
This role query works for sqlite but [fails for Postgres and MySQL](https://github.com/n8n-io/n8n/actions/runs/7269009778/job/19805986017), so generalize by adding alias and accounting for count possibly being `string` in the resulting rows.

Run in progress: https://github.com/n8n-io/n8n/actions/runs/7275986797

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
Can we change [this line](https://github.com/n8n-io/n8n/blob/master/.github/workflows/ci-postgres-mysql.yml#L9) to `packages/cli/src/databases/**` to run DB test on repository and entity changes as well?

File: packages/cli/src/databases/repositories/role.repository.ts [hunk 1]
```diff
@@ -17,17 +17,17 @@ export class RoleRepository extends Repository<Role> {
...
 	async countUsersByRole() {
-		type Row = { role_name: string; count: number };
+		type Row = { role_name: string; count: number | string };
 
 		const rows: Row[] = await this.createQueryBuilder('role')
-			.select('role.name')
+			.select('role.name', 'role_name')
 			.addSelect('COUNT(user.id)', 'count')
 			.innerJoin('user', 'user', 'role.id = user.globalRoleId')
 			.groupBy('role.name')
```
Comment (netroy) [inline packages/cli/src/databases/repositories/role.repository.ts:26]: This will not work with custom schema and table-prefix.

**Review by netroy — CHANGES_REQUESTED** [review]

File: packages/cli/src/databases/repositories/role.repository.ts [hunk 2]
```diff
@@ -17,17 +17,17 @@ export class RoleRepository extends Repository<Role> {
...
 	 */
 	async countUsersByRole() {
-		type Row = { role_name: string; count: number };
+		type Row = { role_name: string; count: number | string };
 
 		const rows: Row[] = await this.createQueryBuilder('role')
-			.select('role.name')
+			.select('role.name', 'role_name')
 			.addSelect('COUNT(user.id)', 'count')
 			.innerJoin('user', 'user', 'role.id = user.globalRoleId')
```
Comment (netroy) [inline packages/cli/src/databases/repositories/role.repository.ts:25]: try this
```suggestion
			.innerJoin(User, 'user', 'role.id = user.globalRoleId')
```

**Review by netroy — CHANGES_REQUESTED** [review]

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.22.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.22.0)

**Pull request merged**